### ISSUE TITLE
Change Request: Support config arrays in plugins that use flat config style

### ISSUE THREAD

**Issue by alecgibson** [eslint/eslint#18752]
### ESLint version

v8.57.0

### What problem do you want to solve?

In "classic config" world, it was possible to define a plugin that had `extends` and `overrides` options:

```js
module.exports = {
  configs: {
    recommended: {
      extends: [/** extends stuff */], 
      overrides: [/** can also override */],
    },
  },
};
```

When I tried to move an existing plugin to use flat config, I first tried to do this using an array for my config:

```js
module.exports = {
  configs: {
    recommended: [ // throws an error - not allowed to be an array
      baseConfig,
      ...overrides,
      // Define new rules
      {...},
    ], 
  },
};
```

Indeed, checking the [documentation](https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-configs-for-flat-config), I see that `recommended` should be defined as an object.

My question is simple: how is a plugin meant to define a config that extends other configs?.

I don't think I'm unusual in wanting to do this. For example, `@typescript-eslint` [extends its own configs](https://github.com/typescript-eslint/typescript-eslint/blob/aba6e37c4050a2031ee4a5340cdbabe25dd5adc4/packages/eslint-plugin/src/configs/all.ts#L11) and will presumably need this feature in order to migrate their configs to flat config?

### What do you think is the correct solution?

Allow plugins to define configs as an array, which is more consistent with the flat config style anyway.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by alecgibson** [comment]
I assume the workaround is to export arrays of configs, and then the consumer would import these arrays using spread syntax, but I don't know if this is necessarily expected or user-friendly syntax:

```js
/** plugin */

module.exports = {
  configGroups: {
    recommended: [],
  },
};
```

```js
/** eslint.config.js */

module.exports = [
  ...plugin.configGroups.recommended,
];
```

**Comment by nzakas** [comment]
Please see the [documention](https://eslint.org/docs/latest/use/configure/combine-configs) on combining configs. If you'd like to add an example using `.flat()`, please feel free.

We did discuss allowing arrays inside of the config array, and even designed and built that feature, but it turned out to create more confusion because instead of wondering if a config was an array (so you could use `.map()` to work with) or an object, you also need to worry if the array contains arrays.

**Comment by alecgibson** [comment]
@nzakas thanks — I assumed it had already been discussed!

I guess my question is now: what's the advice to plugin authors who want their configs to extend other configs? Or is it acceptable to expect consumers to have to spread a plugin config? (I think I'm unclear on what the contract is here)

**Comment by nzakas** [comment]
> I guess my question is now: what's the advice to plugin authors who want their configs to extend other configs? Or is it acceptable to expect consumers to have to spread a plugin config? (I think I'm unclear on what the contract is here)

Export an array and let the consumers spread it. This is [in the plugin docs](https://eslint.org/docs/latest/extend/plugins#configs-in-plugins)

**Comment by alecgibson** [comment]
Aha RTFM 🤦🏼 

Thanks for being patient! Have raised a PR to help future documentationally-challenged people like me https://github.com/eslint/eslint/pull/18753
